R/install.R

```diff
@@ -590,6 +600,46 @@ renv_install_package_impl <- function(record, quiet = TRUE) {
 
 }
 
+renv_install_test <- function(package) {
```

Comment:
This feels like a function that can be called from a vendored renv; does that sound right? I'm not seeing any renv-specific logic/settings that would cause trouble to packrat.

Given that, I'm inclined to abandon my packrat-equivalent implementation and update our vendored renv once this merges.

Any reason to have a separate implementation in packrat?

Should we annotate this function somehow to indicate that it is a semi-public interface?

Reply:
Yep, this should be safe to call from a vendored `renv`.

> Should we annotate this function somehow to indicate that it is a semi-public interface?

Any thoughts on how we should do this? Just a comment, or something slightly more assertive?

Reply:
I don't know that we have much precedent. In the case of packrat, we decided to export `.snapshotImpl` because it was directly called by rsconnect, and having it exported was a reminder that it was part of the public interface.

That said, we don't perfectly follow this pattern. Connect calls `packrat:::appDataDir`, for example, which we never exported. Packrat calls `renv:::download`, which is another example. There is no distinction between "internal" and "public" calls when using the vendored renv.

Mostly, I'm asking for "future you" so we can somehow mark that internal functions are treated as public by the vendoring packages. I don't have any real preference about how to indicate these semi-public functions, but I would hate to end up in a situation where we worry about _all_ of renv effectively being treated as public.